src/player_activity.h

```diff
@@ -4,68 +4,79 @@
 #include "enums.h"
 #include "json.h"
 #include "item_location.h"
+#include "string_id.h"
 
 #include <climits>
 #include <vector>
 
-enum activity_type : int {    // expanded this enum for readability
-    ACT_NULL = 0,
-    ACT_RELOAD,
-    ACT_READ,
-    ACT_GAME,
-    ACT_WAIT,
-    ACT_CRAFT,
-    ACT_LONGCRAFT,
-    ACT_DISASSEMBLE,
-    ACT_BUTCHER,
-    ACT_LONGSALVAGE,
-    ACT_FORAGE,
-    ACT_BUILD,
-    ACT_VEHICLE,
-    ACT_REFILL_VEHICLE, // not used anymore.
-    ACT_TRAIN,
-    ACT_WAIT_WEATHER,
-    ACT_FIRSTAID,
-    ACT_FISH,
-    ACT_PICKAXE,
-    ACT_BURROW,
-    ACT_PULP,
-    ACT_VIBE,
-    ACT_MAKE_ZLAVE,
-    ACT_DROP,
-    ACT_STASH,
-    ACT_PICKUP,
-    ACT_MOVE_ITEMS,
-    ACT_ADV_INVENTORY,
-    ACT_ARMOR_LAYERS,
-    ACT_START_FIRE,
-    ACT_OPEN_GATE,
-    ACT_FILL_LIQUID,
-    ACT_HOTWIRE_CAR,
-    ACT_AIM,
-    ACT_ATM,
-    ACT_START_ENGINES,
-    ACT_OXYTORCH,
-    ACT_CRACKING,
-    ACT_REPAIR_ITEM,
-    ACT_MEND_ITEM,
-    ACT_GUNMOD_ADD,
-    ACT_WAIT_NPC,
-    ACT_CLEAR_RUBBLE,
-    ACT_MEDITATE,
-    NUM_ACTIVITIES
-};
-
 class player;
 class Character;
+class player_activity;
+class activity_type;
+
+using activity_id = string_id<activity_type>;
+
+enum based_on_type : int {
```

Comment:
> In enums.h?

No, just `enum class based_on_type : ...`

> And what kinds of nasty bugs

You're right, not bugs. I mean compile time errors. You cant define e.g. `TIME` anywhere else: the compiler won't be happy. With `class enum`s you can only write `based_on_type::TIME` which is more descriptive and it won't conflict with `whatever_else::TIME`.